### PR TITLE
Random Battles: Refactor Gen 5-7

### PR DESCRIPTION
**Pull request by AnnikaCodes**
As usual, no change in teams from 10,000 different seeds

### CODE REVIEW

File: data/mods/gen6/random-teams.ts [hunk 1]
```diff
@@ -6,6 +6,723 @@ import {toID} from '../../../sim/dex';
...
+	getMediumPriorityItem(
+		ability: string,
+		hasMove: {[k: string]: true},
+		counter: {[k: string]: any},
+		species: Species,
+		moves: ID[],
+		isDoubles: boolean,
+		isLead: boolean
+	): string | undefined {
+		const defensiveStatTotal = species.baseStats.hp + species.baseStats.def + species.baseStats.spd;
```
Comment (ACakeWearingAHat) [inline data/mods/gen6/random-teams.ts:621]:
```suggestion
		const defensiveStatTotal = species.baseStats.hp + species.baseStats.def + species.baseStats.spd;
		const scarfReqs = species.baseStats.spe >= 60 && species.baseStats.spe <= 108 && !counter.priority
```

File: data/mods/gen6/random-teams.ts [hunk 2]
```diff
@@ -6,6 +6,723 @@ import {toID} from '../../../sim/dex';
...
+			) ? 'Choice Scarf' : 'Choice Band';
+		}
+		if (
+			counter.Special >= 4 &&
+			!hasMove['acidspray'] && !hasMove['clearsmog'] && !hasMove['fierydance']
+		) {
+			return (
+				species.baseStats.spa >= 100 &&
+				species.baseStats.spe >= 60 && species.baseStats.spe <= 108 &&
+				!counter.priority && this.randomChance(2, 3)
```
Comment (ACakeWearingAHat) [inline data/mods/gen6/random-teams.ts:644]:
```suggestion
				species.baseStats.spa >= 100 &&
				scarfReqs &&
				this.randomChance(2, 3)
```

File: data/mods/gen6/random-teams.ts [hunk 3]
```diff
@@ -6,6 +6,723 @@ import {toID} from '../../../sim/dex';
...
+			) ? 'Choice Scarf' : 'Choice Specs';
+		}
+		if (
+			counter.Physical >= 3 &&
+			hasMove['defog'] &&
+			species.baseStats.spe >= 60 && species.baseStats.spe <= 108 &&
+			!counter.priority &&
```
Comment (ACakeWearingAHat) [inline data/mods/gen6/random-teams.ts:651]:
```suggestion
			scarfReqs &&
```

File: data/mods/gen6/random-teams.ts [hunk 4]
```diff
@@ -6,6 +6,723 @@ import {toID} from '../../../sim/dex';
...
+			return 'Life Orb';
+		}
+		if (
+			counter.Physical >= 4 &&
+			['bodyslam', 'dragontail', 'fakeout', 'flamecharge', 'rapidspin', 'suckerpunch'].every(m => !hasMove[m])
+		) {
+			return (
+				(species.baseStats.atk >= 100 || ability === 'Huge Power') &&
+				species.baseStats.spe >= 60 && species.baseStats.spe <= 108 &&
+				!counter.priority &&
```
Comment (ACakeWearingAHat) [inline data/mods/gen6/random-teams.ts:633]:
```suggestion
				scarfReqs &&
```

**Review by ACakeWearingAHat — CHANGES_REQUESTED** [review]
just consolidating scarf requirements into a const real quick

**Review by Zarel — APPROVED** [review]
I don't see any problems with this; I'll let you decide how you want to handle this. You don't need my approval for working on random battles, in case that wasn't clear.

**Comment by AnnikaCodes** [comment]
I wasn't awaiting your approval; just making a PR since it's a major change and I want to give others a chance to provide feedback and get more pairs of eyes on the code (which could include you, but in practice is Kris, pyuk, and Cake).

**Pull request merged**